infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesPersistentVolumeClaims.java

```diff
@@ -189,29 +191,14 @@ public PersistentVolumeClaim wait(
               .inNamespace(namespace)
               .withName(name);
 
-      watch =
-          pvcResource.watch(
-              new Watcher<PersistentVolumeClaim>() {
-                @Override
-                public void eventReceived(Action action, PersistentVolumeClaim pvc) {
-                  if (predicate.test(pvc)) {
-                    future.complete(pvc);
-                  }
-                }
-
-                @Override
-                public void onClose(KubernetesClientException cause) {
-                  future.completeExceptionally(
-                      new InfrastructureException(
-                          "Waiting for persistent volume claim '" + name + "' was interrupted"));
-                }
-              });
-
       PersistentVolumeClaim actualPvc = pvcResource.get();
       if (predicate.test(actualPvc)) {
         return actualPvc;
       }
-      try {
+
+      // any of these watchers can finish the operation resolving the future
+      try (Watch boundWatcher = pvcIsBoundWatcher(future, pvcResource, predicate, name);
```

Comment:
Previously, this method was abstract and was used for `waitingBound` only.
Now it's designed to wait for PVC to become bound, please consider renaming method and fixing java docs.
Also, looks like there is no need to pass predicate as an argument anymore.

Reply:
yes, I've removed this method as it was called just from `waitBound` method. So now there is just one and without predicate.